components/camel-test/camel-test-junit5/src/main/java/org/apache/camel/test/junit5/CamelTestSupport.java

```diff
@@ -144,15 +145,16 @@ public void beforeAll(ExtensionContext context) {
             testConfigurationBuilder.withCreateCamelContextPerClass(perClassPresent);
             contextManager = contextManagerFactory.createContextManager(ContextManagerFactory.Type.BEFORE_ALL,
                     testConfigurationBuilder, camelContextConfiguration);
+            ExtensionContext.Store globalStore = context.getStore(ExtensionContext.Namespace.GLOBAL);
+            contextManager.setGlobalStore(globalStore);
```

Comment:
You don't need to change anything right now, but I think may cause some problems for nested tests. In those cases, JUnit may run the before* triggers more than once (and on each of those, we might need to update the global store). 

For now, keep it as a is, but we'll see how it goes after I run a full test of this PR.

Reply:
Ran fine on the CI. Not a problem.